src/NLog.Targets.Syslog/MessageSend/Tcp.cs

```diff
@@ -126,7 +130,7 @@ private Task FramingTask(ByteArray message)
 
             var octetCount = message.Length;
             var prefix = new ASCIIEncoding().GetBytes($"{octetCount} ");
-            return Task.Factory.SafeFromAsync(stream.BeginWrite, stream.EndWrite, prefix, 0, prefix.Length, null);
+            return stream.WriteAsync(prefix, 0, prefix.Length);
```

Comment:
WriteAsync internally relies on Task.Factory.FromAsync which does not create a faulted Task when a Socket exceptions occurs: that's why SafeFromAsync should be kept